src/cpp/client.cpp

```diff
@@ -702,21 +696,41 @@ parsed_reply_map Client::get_db_cluster_info(std::string address)
     if (host.empty() or port == 0)
         throw std::runtime_error(std::string(address) +
                                  "is not a valid database node address.");
-    Command cmd;
+    ClusterInfoCommand cmd;
     cmd.set_exec_address_port(host, port);
     cmd.add_field("CLUSTER");
     cmd.add_field("INFO");
     CommandReply reply = this->_run(cmd);
-    return CommandReplyParser::parse_db_cluster_info(std::string(reply.str(),
+    return cmd.parse_db_cluster_info(std::string(reply.str(),
                                                      reply.str_len()));
 }
 
-CommandReply Client::_run(Command& cmd)
+inline CommandReply Client::_run(AddressAtCommand& cmd)
+{
+    return this->_redis_server->run(cmd);
+}
+
+inline CommandReply Client::_run(AddressAnyCommand& cmd)
 {
     return this->_redis_server->run(cmd);
 }
 
-CommandReply Client::_run(CommandList& cmds)
+inline CommandReply Client::_run(SingleKeyCommand& cmd)
+{
+    return this->_redis_server->run(cmd);
+}
+
+inline CommandReply Client::_run(MultiKeyCommand& cmd)
+{
+    return this->_redis_server->run(cmd);
+}
+
+inline CommandReply Client::_run(CompoundCommand& cmd)
+{
+    return this->_redis_server->run(cmd);
+}
+
+inline std::vector<CommandReply> Client::_run(CommandList& cmds)
```

Comment:
If these are inlines, then they should be implemented in the header, not in the cpp file. Either remove inline or move the implementation